### PR TITLE
44 tree components for resources page side menu

### LINKED ISSUE #44: Tree components for Resources page side menu
The Resources page side menu is long and might look better with collapsible elements: [Shoelace tree component](https://shoelace.style/components/tree) does exactly this.

image
to 
image

However, @atruskie tested this and there was some issue with showing the current page correctly. 

- [x] Implement the tree component
- [x] Investigate how to show the current page on the menu correctly

### PR DESCRIPTION
**Pull request by nellihol**
Long titles look a bit funny, but okay:
image

Closes #44

### CODE REVIEW

File: assets/css/resources.scss [hunk 1]
```diff
@@ -46,6 +46,33 @@
...
+        }
+
+        a {
+            font-size: 0.9em;
+            white-space: normal;
+            line-height: 1;
+            font-family: roboto,sans-serif;
+
+            &:hover {
+                opacity: .5;
```
Comment (atruskie) [inline assets/css/resources.scss:71]: just double check we can't use any of our style variables for these (--oe-padding, --oe-font ?)

File: themes/hugo-book-custom/layouts/partials/docs/menu-filetree.html [hunk 2]
```diff
@@ -4,25 +4,52 @@
...
+    {{ template "book-section-children" (dict "Section" . "CurrentPage" $) }}
+    </sl-tree>
+
+    <script>
+    document.addEventListener("DOMContentLoaded", function () {
+        const activeLink = document.querySelector("sl-tree-item a.active");
+
+        var getParents = function (elem, selector) {
+	        var parents = [];
+	        for ( ; elem && elem !== document; elem = elem.parentNode ) {
```
Comment (atruskie) [inline themes/hugo-book-custom/layouts/partials/docs/menu-filetree.html:17]: just make this a while loop

File: themes/hugo-book-custom/layouts/partials/docs/menu-filetree.html [hunk 3]
```diff
@@ -4,25 +4,52 @@
...
+		        if (selector) {
+			        if (elem.matches(selector)) {
+				        parents.push(elem);
+			        }
+			        continue;
+		        }
+		        parents.push(elem);
+	        }
+	        return parents;
```
Comment (atruskie) [inline themes/hugo-book-custom/layouts/partials/docs/menu-filetree.html:26]: you seem to be supporting two cases here; with a selector, and without. Just make the function only support a selector, seems easier.

**Review by atruskie — CHANGES_REQUESTED** [review]
Ok while testing I found the behavior a little odd.

- the select event has a different style from the active style (left blue solid bar, and transparent background color)
- the result is selection looks different after page load
- it may look better if the selected style is applied to the deepest child, and this expanded style is applied to the parents
- there's a weird flash of unstyled content showed while loading between pages
    - perhaps we should set some basic custom styles for the shoelace elements that are shown while we load shoelace and while we wait for your recursive selector to run.

[tree selection.webm](https://user-images.githubusercontent.com/1452968/206949249-7bab9f89-e483-410d-8d41-5c71d6b12466.webm)

image

Otherwise, I look forward to this change, it should be super cool looking!

**Comment by atruskie** [comment]
Another thought, for maybe another issue: perhaps we can support short titles for really long titles, either:

- with a short title front matter item (manual)
- or some kind of truncation on titles that are too long (manual)

**Comment by andrew-1234** [comment]
> * there's a weird flash of unstyled content showed while loading between pages

I just noticed this on the live website too, it looks like the `sl-drawer` element, flashing contact, governance, resources.

Adding this to main_nav.css seems to stop it:

```
.nav-drawer:not(:defined) {
    visibility: hidden;
  }
```

**Review by andrew-1234 — COMMENTED** [review]
I really like it, makes the site look much cleaner! Agree with changing how the `selected` attribute (box with blue line) functions. When clicking on deepest child pages, seems there is no need for it to appear at all if it will just disappear again?

**Comment by andrew-1234** [comment]
I also just noticed that the `selected` element looks a bit weird and janky when changing pages in safari. Looking through the shoelace change logs safari bugs seem to be fairly common, so dealing with safari visual bugs is probably worth a separate issue. 

Screen Shot 2022-12-13 at 9 44 40 am

File: themes/hugo-book-custom/layouts/partials/docs/menu-filetree.html (hunk 2 above)
Comment (nellihol) [reply themes/hugo-book-custom/layouts/partials/docs/menu-filetree.html]: Done!

File: themes/hugo-book-custom/layouts/partials/docs/menu-filetree.html (hunk 3 above)
Comment (nellihol) [reply themes/hugo-book-custom/layouts/partials/docs/menu-filetree.html]: modified

**Comment by nellihol** [comment]
Focus style is now: 
image

**Pull request merged**